src/shieldpy/automata/visualize.py

```diff
@@ -0,0 +1,20 @@
+from shieldpy.automata.nondeterministic_finite import NFA, Transition
+from visual_automata.fa.nfa import VisualNFA
+
+def visualize_nfa(nfa: NFA):
+    """Visualize the NFA using an automata visualization library.
+    """
+
+    nfa = VisualNFA(
+        states={s.name for s in nfa.states},
+        # TODO maybe need merge operator on all match t.start.name for the format they have?
+        transitions= {
+            t.start.name: {
+                t.symbol.name: {t.end.name}
+            } for t in nfa.transitions
+        }, # TODO
+        initial_state=nfa.start.name,
+        final_states={s for s in nfa.accept},
+        input_symbols={a.name for a in nfa.alphabet}
+    )
```

Comment:
The expected format should be something like this:

```python
VisualNFA(
    states={"q0", "q1", "q2"},
    input_symbols={"0", "1"},
    transitions={
        "q0": {"": {"q2"}, "1": {"q1"}},
        "q1": {"1": {"q2"}, "0": {"q0", "q2"}},
        "q2": {},
    },
    initial_state="q0",
    final_states={"q0"},
)
```

So the mapping should be pretty simple.

Reply:
i think it's 

```python
{
            ttt.start.name: {
                tt.symbol.name: {t.end.name for t in nfa.transitions if t.start.name == ttt.start.name and t.symbol.name == tt.symbol.name} for tt in nfa.transitions if tt.start.name == ttt.start.name
            } for ttt in nfa.transitions
        }
```